src/language-markdown/embed.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+const util = require("../common/util");
+const doc = require("../doc");
+const docBuilders = doc.builders;
+const hardline = docBuilders.hardline;
+const concat = docBuilders.concat;
+
+function embed(path, print, format, options) {
+  const node = path.getValue();
+
+  if (node.type === "code") {
+    const parser = getParserName(node.lang);
+    if (parser) {
+      const styleUnit = options.__inJsTemplate ? "~" : "`";
+      const style = styleUnit.repeat(
+        Math.max(3, util.getMaxContinuousCount(node.value, styleUnit) + 1)
+      );
+      const doc = format(node.value, { parser }, options);
+      return concat([style, node.lang, hardline, doc, style]);
+    }
+  }
+
+  return null;
+
+  function getParserName(lang) {
```

Comment:
It seems we can re-use the support info (langName/extName) to get the correct parser? and it'll work with all external plugins.

Reply:
Good idea! Done.